### PR TITLE
Data processing pipeline

### PR DESCRIPTION
**Pull request by esheehan-gsl**
Create a Lambda function that can process the incoming NetCDF diag files into a
Zarr array in S3 for the application to read.

### CODE REVIEW

**Comment by esheehan-gsl** [comment]
@ian-noaa I think we’re ready to test the pipeline for real. Aside from an IAM policy that allows the Lambda function to read from Ed’s S3 bucket with the NetCDF files, and permissions to write to our S3 bucket, we’ll need to add an environment variable — `UG_UPLOAD_BUCKET` — that contains the `s3://` URL for our bucket, where we write the Zarr file.

**Comment by esheehan-gsl** [comment]
I think I spoke too soon…

**Comment by esheehan-gsl** [comment]
@ian-noaa ok, I think this is now actually ready to test in AWS. I had to update the file handling to account for the fact that the diag files in the S3 bucket have a slightly different naming convention than the test data we've been working with, and that they are gzipped.

We’ll need to set the `UG_UPLOAD_BUCKET` environment variable to the S3 bucket where we want our Zarr stored, and the Lambda function will need permission to read from the parallelcluster-rtma-cluster bucket.

**Comment by esheehan-gsl** [comment]
Shoot. Found one more bug.

**Comment by esheehan-gsl** [comment]
@ian-noaa I think we’re ready to go on this. We can test the Lambda function <i>in situ</i> now. I ran it locally, and it appears to be working.

We need to set `UG_DIAG_ZARR` in the environment as a path to where we want the Zarr in S3. Could be something like `s3://unified-graphics/diagnostics.zarr` or something. We need to make sure we don’t try to run anything from before 202202021400 because those NetCDF files are all empty.

File: services/data/src/ugdata/aws.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    download_path : str
+        The location to which the file will be downloaded (default is "/tmp")
+
+    Returns
+    -------
+    pathlib.Path
+        The path to the downloaded file
+    """
+    tmp_key = key.replace("/", "_")
+    tmp_path = Path(download_path) / f"{uuid.uuid4()}-{tmp_key}"
```
Comment (ian-noaa) [inline services/data/src/ugdata/aws.py:33]: Using a UUID is a really nice way to make sure files don't conflict.

**Review by ian-noaa — APPROVED** [review]
This looks good to me.

VLab is still working on testing the Lambda function. Right now they're sorting out the various permissions Lambda needs to run. I'll let you know when they've had a successful run. Longer term, they're going to setup some deployment stuff on their end to update the Lambda whenever the `main` image changes.

**Pull request merged**